protocol/x/clob/keeper/equity_tier_limit.go

```diff
@@ -59,9 +59,12 @@ func (k *Keeper) InitializeEquityTierLimit(
 // tier limits enforce.
 //
 // Note that the method is dependent on whether we are executing on `checkState` or on `deliverState` for
-// stateful orders. During `checkState` we rely on the uncommitted order count to tell us how many stateful
-// orders exist outside of the MemClob. For `deliverState` we use the `ProcessProposerMatchesEvents`
-// to find out how many orders (minus removals) exists outside of the `MemClob`.
+// stateful orders. For `deliverState`, we sum:
+//   - the number of long term orders.
+//   - the number of triggered conditional orders.
+//   - the number of untriggered conditional orders.
+//
+// And for `checkState`, we add to the above sum the number of to be committed stateful orders.
 func (k Keeper) ValidateSubaccountEquityTierLimitForNewOrder(ctx sdk.Context, order types.Order) error {
```

Comment:
Small suggestion on simplifying the loop between lines 105 - 112 (IMO slightly more readable because no else statement):
```
	for _, limit := range equityTierLimits {
		if netCollateral.Cmp(limit.UsdTncRequired.BigInt()) < 0 {
			break
		}
                 equityTierLimit = limit
	}
```